pkgs/development/misc/newlib/default.nix

```diff
@@ -45,6 +54,18 @@ stdenv.mkDerivation rec {
 
   dontDisableStatic = true;
 
+  # apply necessary nano changes from https://developer.arm.com/-/media/Files/downloads/gnu/12.2.rel1/manifest/copy_nano_libraries.sh?rev=4c50be6ccb9c4205a5262a3925317073&hash=1375A7B0A1CD0DB9B9EB0D2B574ADF66
+  postInstall = lib.optionalString nanoizeNewlib ''
+    mkdir -p $out/${stdenv.targetPlatform.config}/include/newlib-nano
+    cp $out/${stdenv.targetPlatform.config}/include/newlib.h $out/${stdenv.hostPlatform.config}/include/newlib-nano/
+
+    cd $out/${stdenv.targetPlatform.config}/lib
```

Comment:
Is it safe to `cd` in a hook and not restore the default directory here?

Reply:
changed to a subshell